### PR TITLE
Allow setting of Game to nil, and allow the game type to be "listening"

### PR DESCRIPTION
**Pull request by dannyt66**
This code serves to allow the library to set the user's Game to nil, or not currently playing a game, and also allows for the use of Discord's new "Listening to" function to be used in place of "Playing" and "Streaming".

### CODE REVIEW

**Comment by dannyt66** [comment]
This code is breaking the build as it breaks compatibility with the old UpdateStatus method - I'd propose fixing this by having two instances of the method, one which uses the old system, however I do not know if this is acceptable to the way this library maintains compatibility. Please advise on the best fix for this.

File: wsapi.go [hunk 1]
```diff
@@ -328,7 +328,7 @@ type updateStatusOp struct {
 // If game!="" then set game.
 // If game!="" and url!="" then set the status type to streaming with the URL set.
 // if otherwise, set status to active, and no game.
-func (s *Session) UpdateStreamingStatus(idle int, game string, url string) (err error) {
+func (s *Session) UpdateStreamingStatus(idle int, game string, url string, isSong bool) (err error) {
```
Comment (iopred) [inline wsapi.go]: This will break existing users of the API, I think just the change to line 337 is enough to allow this.

If you want a simple way to update the status, I suggest just adding an 'UpdateListeningStatus' method, but I would shy away from that personally.

Comment (dannyt66) [reply wsapi.go]: Branch changed, apoligies. I think the ability to send the listening status should be added, however I'm not sure how you'd best want me to add this if not by adding an 'UpdateListeningStatus' method.

Comment (iopred) [reply wsapi.go]: UpdateStatusComplex would be the way to call this, adding some additional comments on the method could explain this to users.

**Comment by dannyt66** [comment]
New commits pushed to maintain compatibility, whilst still allowing the update of listening status.

File: discord.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 )
 
 // VERSION of DiscordGo, follows Semantic Versioning. (http://semver.org/)
-const VERSION = "0.18.0-alpha"
+const VERSION = "0.18.0"
```
Comment (iopred) [inline discord.go]: please undo this change.

File: wsapi.go [hunk 3]
```diff
@@ -351,6 +351,9 @@ func (s *Session) UpdateStreamingStatus(idle int, game string, url string) (err
 			URL:  url,
 		}
 	}
+	if game == "" {
+		usd.Game = nil
```
Comment (iopred) [inline wsapi.go]: This shouldn't be needed, the default is nil.

File: wsapi.go [hunk 4]
```diff
@@ -379,6 +382,22 @@ func (s *Session) UpdateStatus(idle int, game string) (err error) {
...
+//If game!="" then set to what user is listening to
+//Else, set user to active and no game.
+func (s *Session) UpdateListeningStatus(idle int, game string) (err error) {
+	gameType := GameTypeListening
+	usd := UpdateStatusData{
+		Status: "online",
+	}
+	usd.Game = &Game{
+		Name: game,
+		Type: gameType,
```
Comment (iopred) [inline wsapi.go]: This probably reads a bit better as:


return s.UpdateStatusComplex(UpdateStatusData{
  Status: "online",
  Game: &Game {
    Name: game,
    Type: GameTypeListening,
    URL: "",
  }
})

File: wsapi.go (hunk 3 above)
Comment (dannyt66) [reply wsapi.go]: This is definitely not the case, if I've sent "" without this change, the game has been left as it was previously set to.

File: discord.go (hunk 2 above)
Comment (dannyt66) [reply discord.go]: Sorry, this change was inadvertent. Will fix.

File: wsapi.go (hunk 3 above)
Comment (iopred) [reply wsapi.go]: Hmmmmm, I think I see what's going on, this check should be in UpdateListeningStatus then, sorry for the back and forth on this one!

File: wsapi.go [hunk 5]
```diff
@@ -379,6 +382,20 @@ func (s *Session) UpdateStatus(idle int, game string) (err error) {
 	return s.UpdateStreamingStatus(idle, game, "")
 }
 
+//UpdateListeningStatus is used to set the user to "Listening to..."
+//If game!="" then set to what user is listening to
+//Else, set user to active and no game.
+func (s *Session) UpdateListeningStatus(idle int, game string) (err error) {
+	return s.UpdateStatusComplex(UpdateStatusData{
```
Comment (iopred) [inline wsapi.go]: Sorry, I was misunderstanding what you're trying here, this should be:

var g Game
if game != "" {
  g = &Game{
    Name: game,
    Type: GameTypeListening,
    URL:  "",
  }
}
return s.UpdateStatusComplex(UpdateStatusData{
  Status: "online",
  Game: g,
})

File: wsapi.go (hunk 3 above)
Comment (dannyt66) [reply wsapi.go]: This behaviour exhibits itself when setting an actual game too, which is why I put the check in there. I do agree that I probably need a similar check in UpdateListeningStatus as well, however.

File: wsapi.go [hunk 6]
```diff
@@ -379,6 +379,23 @@ func (s *Session) UpdateStatus(idle int, game string) (err error) {
 	return s.UpdateStreamingStatus(idle, game, "")
 }
 
+//UpdateListeningStatus is used to set the user to "Listening to..."
+//If game!="" then set to what user is listening to
+//Else, set user to active and no game.
+func (s *Session) UpdateListeningStatus(idle int, game string) (err error) {
```
Comment (iopred) [inline wsapi.go]: Let's make this:

UpdateListeningStatus(game)

Idle seems kinda useless and people can do it with Complex.

File: wsapi.go [hunk 7]
```diff
@@ -379,6 +379,23 @@ func (s *Session) UpdateStatus(idle int, game string) (err error) {
 	return s.UpdateStreamingStatus(idle, game, "")
 }
 
+//UpdateListeningStatus is used to set the user to "Listening to..."
```
Comment (iopred) [inline wsapi.go]: Could you put a space between // and the comments on these lines.
